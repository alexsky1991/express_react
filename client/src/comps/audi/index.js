import Audi from './audi'

export default Audi