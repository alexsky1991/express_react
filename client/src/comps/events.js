import {EventEmitter} from 'events';

let voteEvent=new EventEmitter(); 

export {voteEvent};
