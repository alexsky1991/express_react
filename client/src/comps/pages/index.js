import FirstPage from './first-page';
import GamePage from './game-page';
import AdminPage from './admin-page';

export {FirstPage, GamePage, AdminPage};