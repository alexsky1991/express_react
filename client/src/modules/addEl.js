
const addEl = (item, arr) => {
	let new_arr = [...arr];
	new_arr.unshift(item);

	return new_arr;
}

export {addEl}