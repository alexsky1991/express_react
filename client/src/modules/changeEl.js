
const changeEl = (item, arr) => {

	let new_arr = [...arr];

	new_arr = new_arr.map(el => {
		if(el.id === item.id)
			el = item

		return el
	})

	return new_arr

}

export {changeEl}

