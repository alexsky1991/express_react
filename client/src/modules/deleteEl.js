
const deleteEl = (id, arr) => {

	let new_arr = [...arr];

	new_arr = new_arr.filter(el => el.id !== id);

	return new_arr;
}

export {deleteEl}