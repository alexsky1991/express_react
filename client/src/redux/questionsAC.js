const QUESTIONS_LOADING='QUESTIONS_LOADING';
const QUESTIONS_ERROR='QUESTIONS_ERROR';
const QUESTIONS_SET='QUESTIONS_SET';

const questionsLoadingAC=function() {
  return {
    type: QUESTIONS_LOADING,
  };
}

const questionsErrorAC=function() {
  return {
    type: QUESTIONS_ERROR,
  };
}

const questionsSetAC=function(questions) {
  return {
    type: QUESTIONS_SET,
    questions:questions,
  };
}

export {
  questionsLoadingAC,QUESTIONS_LOADING,
  questionsErrorAC,QUESTIONS_ERROR,
  questionsSetAC,QUESTIONS_SET,
}
