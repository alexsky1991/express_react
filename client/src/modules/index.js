import {addEl} from './addEl';
import {deleteEl} from './deleteEl';
import {changeEl} from './changeEl';

export { addEl, deleteEl, changeEl}