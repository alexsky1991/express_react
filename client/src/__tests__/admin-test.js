"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import {AdminPage} from '../comps/pages';
import Item from '../comps/item'

import {createStore} from 'redux';
import {connect} from 'react-redux';


import {addEl, deleteEl,changeEl} from '../modules';

import { questionsLoadingAC, questionsErrorAC, questionsSetAC } from "../redux/questionsAC";

import combinedReducer from '../redux/reducers.js';
let store=createStore(combinedReducer);


test('работа adminPage', () => {
	const match = {params: {clid:1}}

    const component = renderer.create(
      <AdminPage match={match}/>
    );


  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const buttonAdd = component.root.find( el => el.props.className=='admin_new' ); 
 
  buttonAdd.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

});

test('работа Item', () => {

  const el = {
  	question:'C каким из этих государств Россия не имеет сухопутной границы?',
    answers: ["Таджикистан", "Норвегия", "Польша", "КНДР"],
    id:200222
  };

  const component = renderer.create(
    <Item key={el.id} item={el} idx={1}/>
  );


  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const buttonEdit = component.root.find( el => el.props.className=='item_edit' ); 
 
  buttonEdit.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();


});