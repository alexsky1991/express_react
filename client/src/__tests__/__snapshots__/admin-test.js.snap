// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Item 1`] = `
<div
  className="item"
  style={
    Object {
      "animationDelay": "0.05s",
    }
  }
>
  <div
    className="item_panel"
  >
    <div
      className="item_question"
    >
      C каким из этих государств Россия не имеет сухопутной границы?
    </div>
    <div
      className="item_edit"
      onClick={[Function]}
    >
      редактировать
    </div>
  </div>
  <div
    className="item_answers"
  >
    <div
      className="item_answer"
    >
      Таджикистан
    </div>
    <div
      className="item_answer"
    >
      Норвегия
    </div>
    <div
      className="item_answer"
    >
      Польша
    </div>
    <div
      className="item_answer"
    >
      КНДР
    </div>
  </div>
</div>
`;

exports[`работа Item 2`] = `
<div
  className="item item_bg_edit"
  style={
    Object {
      "background": "#7b684a",
    }
  }
>
  <div
    className="item_panel"
  >
    <div
      className="item_question"
    >
      <textarea
        className="item_question_input"
        defaultValue="C каким из этих государств Россия не имеет сухопутной границы?"
        onBlur={[Function]}
        size={62}
        type="text"
      />
    </div>
    <div
      className="item_right"
    >
      <div
        className="item_delete"
        onClick={[Function]}
      >
        удалить
      </div>
      <div
        className="item_edit"
        onClick={[Function]}
      >
        редактировать
      </div>
    </div>
  </div>
  <div
    className="item_answers"
  >
    <input
      className="item_answer_input"
      defaultValue="Таджикистан"
      number={0}
      onBlur={[Function]}
      style={
        Object {
          "width": "108px",
        }
      }
    />
    <input
      className="item_answer_input"
      defaultValue="Норвегия"
      number={1}
      onBlur={[Function]}
      style={
        Object {
          "width": "81px",
        }
      }
    />
    <input
      className="item_answer_input"
      defaultValue="Польша"
      number={2}
      onBlur={[Function]}
      style={
        Object {
          "width": "63px",
        }
      }
    />
    <input
      className="item_answer_input"
      defaultValue="КНДР"
      number={3}
      onBlur={[Function]}
      style={
        Object {
          "width": "45px",
        }
      }
    />
  </div>
</div>
`;

exports[`работа adminPage 1`] = `
<div
  className="admin"
>
  <div
    className="admin_header"
  >
    <div
      className="admin_header_amount"
    >
      Количество вопросов: 
      0
    </div>
    <div
      className="admin_title"
    >
      Cписок вопросов:
    </div>
    <div
      className="admin_new"
      onClick={[Function]}
    >
      Добавить вопрос
    </div>
  </div>
  <div
    className="admin_list"
  />
  <div
    className="admin_navigation"
    onClick={[Function]}
  />
</div>
`;

exports[`работа adminPage 2`] = `
<div
  className="admin"
>
  <div
    className="admin_header"
  >
    <div
      className="admin_header_amount"
    >
      Количество вопросов: 
      0
    </div>
    <div
      className="admin_title"
    >
      Cписок вопросов:
    </div>
    <div
      className="admin_new"
      onClick={[Function]}
    >
      Добавить вопрос
    </div>
  </div>
  <div
    className="item new_item"
    style={
      Object {
        "background": "#7b684a",
      }
    }
  >
    <div
      className="item_panel"
    >
      <div
        className="item_question"
      >
        <input
          className="item_question_input"
          id="new_question"
          placeholder="новый вопрос"
          size="50"
          type="text"
        />
      </div>
      <div
        className="item_right"
      >
        <div
          className="item_delete"
          onClick={[Function]}
        >
          удалить
        </div>
        <div
          className="item_edit"
          onClick={[Function]}
        >
          добавить
        </div>
      </div>
    </div>
    <div
      className="item_answers"
      id="item_answers"
    >
      <input
        className="item_answer_input"
        placeholder="вариант 1"
        type="text"
      />
      <input
        className="item_answer_input"
        placeholder="вариант 2"
        type="text"
      />
      <input
        className="item_answer_input"
        placeholder="вариант 3"
        type="text"
      />
      <input
        className="item_answer_input"
        placeholder="вариант 4"
        type="text"
      />
    </div>
  </div>
  <div
    className="admin_list"
  />
  <div
    className="admin_navigation"
    onClick={[Function]}
  />
</div>
`;
